pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,489 @@
+                \n  SRQ for Cal err: {}\n  SQR for Power on: {}\
+                \nError information: \n Calibration: {}  RAM: {}  ROM: {}  AD Slope: {} AD: {} AD-Link: {} \
+                \nDAC value: {}".format(
+                cur_mode, r_str, 6-self.digits, self.int_trig, self.ext_trig, self.auto_range,
+                self.auto_zero, self.fifty_hz, self.front_rear , self.cal_enable,
+                self.SRQ_data_rdy, self.SRQ_syntax_error, self.SRQ_internal_error, self.SRQ_front_panel, self.SRQ_cal_error, self.SRQ_PON,
+                self.ERR_cal, self.ERR_RAM, self.ERR_ROM, self.ERR_slope, self.ERR_AD, self.ERR_AD_Link,
+                self.DAC_value)
+            )
+
+class Status(ctypes.Union):
+    """Union type element for the decoding of the status bit-fields
+    """
+    _fields_ = [
+        ("B", Status_bytes),
+        ("b", Status_bits)
+    ]
+
+
+
+class HP3478A(Instrument):
+    """ Represents the Hewlett Packard 3748A 5 1/2 digit multimeter
+    and provides a high-level interface for interacting
+    with the instrument.
+
+    As this unit predates SCPI some tricks are required to get this working
+
+    """
+
+    def __init__(self, resourceName, **kwargs):
+        super(HP3478A, self).__init__(
+            resourceName,
+            "Hewlett-Packard HP3478A",
+            includeSCPI = False,
+            send_end = True,
+            read_termination = "\r\n",
+            **kwargs
+        )
+
+    # Definitions for different specifics of this instrument
+    MODES={"DCV": "F1",
+           "ACV": "F2",
+           "R2W": "F3",
+           "R4W": "F4",
+           "DCI": "F5",
+           "ACI": "F6",
+           "Rext": "F7",
+           }
+
+    RANGES={"DCV": {3E-2: "R-2", 3E-1: "R-1",3: "R0",30: "R1",300:"R2",
+                    "auto": "RA"},
+            "ACV": {3E-1: "R-1", 3: "R0",30: "R1" ,300: "R2","auto": "RA"},
+            "R2W": {30: "R1", 300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5",
+                    3E6: "R6", 3E7: "R7","auto": "RA"},
+            "R4W": {30: "R1",300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5",
+                    3E6: "R6", 3E7: "R7","auto": "RA"},
+            "DCI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "ACI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "Rext": {3E7: "R7","auto": "RA"},
+            }
+
+    TRIGGERS={
+            "auto": "T1",
+            "internal": "T1",
+            "external": "T2",
+            "single": "T3",
+            "hold": "T4",
+            "fast": "T5",
+            }
+
+    ERRORS={
+            1: "Calibration error",
+            2: "RAM error",
+            4: "ROM error",
+            8: "AD slope error",
+            16: "AD converter error",
+            32: "AD link error",
+           }
+    
+
+
+    #decoder functions
+    @classmethod
+    def decode_status(self,status_bytes, field=None):
+        """Method to handle the decoding of the status bytes into something meaningfull
```

Comment:
If you replace `@classmethod` with `@staticmethod` and remove the `self` parameter, you should be good to go with static method.